### PR TITLE
MC-35016: Out of stock products doesn't filter properly using "price" filter

### PR DESCRIPTION
**Pull request by tsviklinskyi**
### Related Pull Requests
- https://github.com/magento-tsg-csl3/magento2ce/pull/515

### Description (*)

ISSUE
Out of stock products doesn't filter properly using "price" filter.

CONFIGURATIONS
01. Checked with inventory modules
02. Display Out of Stock Products > enabled
03. Elasticsearch6 enabled

STEPS TO REPRODUCE
01. Create a configurable product with two simple products (set price as $1500)
02. Both simple products should "out of stock" while creating the configurable product
03. Assign this configurable product to a category
04. Reindex and check "catalog_product_index_price" table using entity id of the above configurable product
05. All the prices are save as 0
06. Load the category and confirm the availability of the product
07. Open the "price" filter from layered navigation
08. Notice that it has an option as [ $0.00 - $9.99 ]
09. Click on this option and you will get the configurable product we created before by setting the price as $1500

EXPECTED RESULT
The product should filter under the correct price range.

ACTUAL RESULT
the product falls under wrong price range filter.

NOTE
It seems this happens due to the 0 value assigned to configurable product in "catalog_product_index_price" table. Apart from the above scenario, there is another possibility where this could occur which described in detail under MC-33010. But that was closed without fixing due to unacceptable performance degradation levels.

The current merchant attached to this ticket has products created with both the scenarios. Please let us know how to proceed on this. Due to that their "price" filtering doesn't work as expected.

### Fixed Issues (if relevant)

1. Fixes magento/inventory#<issue_number>: Issue title

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by tsviklinskyi** [comment]
@magento run all tests

**Review by sidolov — DISMISSED** [review]

**Comment by tsviklinskyi** [comment]
@magento run all tests

**Comment by tsviklinskyi** [comment]
@magento run all tests

**Comment by tsviklinskyi** [comment]
@magento run all tests

**Comment by tsviklinskyi** [comment]
@magento run all tests

File: InventoryConfigurableProduct/Pricing/Price/Indexer/StockStatusBaseSelectProcessor.php [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+use Magento\InventoryCatalogApi\Api\DefaultStockProviderInterface;
+use Magento\InventoryIndexer\Model\StockIndexTableNameResolverInterface;
+use Magento\InventorySalesApi\Api\Data\SalesChannelInterface;
+use Magento\InventorySalesApi\Api\StockResolverInterface;
+use Magento\Store\Model\StoreManagerInterface;
+
+/**
+ * Base select processor.
+ */
+class StockStatusBaseSelectProcessor implements BaseSelectProcessorInterface
```
Comment (sidolov) [inline InventoryConfigurableProduct/Pricing/Price/Indexer/StockStatusBaseSelectProcessor.php]: Let's rename it to `BaseStockStatusSelect`

**Review by sidolov — CHANGES_REQUESTED** [review]

File: InventoryConfigurableProduct/Pricing/Price/Indexer/StockStatusBaseSelectProcessor.php (hunk 1 above)
Comment (tsviklinskyi) [reply InventoryConfigurableProduct/Pricing/Price/Indexer/StockStatusBaseSelectProcessor.php]: Done

**Comment by tsviklinskyi** [comment]
@magento run all tests

**Comment by tsviklinskyi** [comment]
@magento run integration

**Comment by tsviklinskyi** [comment]
@magento run all tests

**Comment by tsviklinskyi** [comment]
@magento run Functional Tests EE

**Comment by tsviklinskyi** [comment]
@magento run all tests

**Comment by tsviklinskyi** [comment]
@magento run Functional Tests CE
@magento run Integration Tests 
@magento run WebAPI Tests

**Comment by tsviklinskyi** [comment]
@magento run Integration
@magento run WebAPI

**Comment by tsviklinskyi** [comment]
@magento run all tests

**Comment by tsviklinskyi** [comment]
@magento run Functional Tests EE

**Review by sidolov — APPROVED** [review]

**Comment by sidolov** [comment]
@magento import pr to magento-commerce/inventory

**Comment by magento-engcom-team** [comment]
@sidolov the pull request successfully imported.

**Pull request closed without merging**